File: c# Dotnet Core/STED/src/DataProviders/UserByNameLoader.cs
using Domain;
using UseCases;

namespace DataProviders
{
    public class UserByNameLoader : IUserByNameLoader
    {
        public IUser? Load(string name)
        {
            if (name == null)
                return null;
            if (name == "dexter")
                return new User { Name = "dexter", Password = "killer" };

            return null;
        }
    }
}

File: Go/STED/DataProviders/UserByNameLoader.go
package DataProviders

import "serenityapi/Domain"

type UserByNameLoader struct{}

func (l *UserByNameLoader) Load(name string) *Domain.User {
	if name == "dexter" {
		return &Domain.User{Name: "dexter", Password: "killer"}
	}
	return nil
}
